main.c

```diff
@@ -255,57 +343,76 @@ static int dbus_event(int fd, uint32_t mask, void *data) {
 	return count;
 }
 
-static void connect_to_bus(void) {
-	int ret = sd_bus_default_system(&bus);
+static void set_session(void) {
 	sd_bus_message *msg = NULL;
 	sd_bus_error error = SD_BUS_ERROR_NULL;
-	pid_t my_pid = getpid();
 	const char *session_name_tmp;
-	if (ret < 0) {
-		errno = -ret;
-		swayidle_log_errno(LOG_ERROR, "Failed to open D-Bus connection");
-		return;
-	}
-	struct wl_event_source *source = wl_event_loop_add_fd(state.event_loop,
-		sd_bus_get_fd(bus), WL_EVENT_READABLE, dbus_event, bus);
-	wl_event_source_check(source);
-	ret = sd_bus_call_method(bus, "org.freedesktop.login1",
-			"/org/freedesktop/login1",
-			"org.freedesktop.login1.Manager", "GetSessionByPID",
+	pid_t my_pid = getpid();
+
+	int ret = sd_bus_call_method(bus, DBUS_LOGIND_SERVICE, DBUS_LOGIND_PATH,
+			DBUS_LOGIND_MANAGER_INTERFACE, "GetSessionByPID",
 			&error, &msg, "u", my_pid);
 	if (ret < 0) {
-		swayidle_log(LOG_ERROR,
-				"Failed to find session name: %s", error.message);
-		goto cleanup;
+		swayidle_log(LOG_DEBUG,
+				"Failed to find session using PID: %s", error.message);
+		sd_bus_error_free(&error);
+		sd_bus_message_unref(msg);
+
+		const char *env_session = getenv("XDG_SESSION_ID");
+		ret = sd_bus_call_method(bus, DBUS_LOGIND_SERVICE, DBUS_LOGIND_PATH,
```

Comment:
Randomly stumbled upon this PR while troubleshooting an issue with IdleHint not working when swayidle is started as a systemd user service.

Have you considered using `auto` magic session name instead of looking up the ID in the environment? That's what [Gnome 3.34 is doing](https://blogs.gnome.org/benzea/2019/10/01/gnome-3-34-is-now-managed-using-systemd/)

That said, this is way outside of my area of expertise so feel free to completely ignore the comment.

Reply:
@melentye Thanks for the suggestion: I like this approach more than pulling from the environment. The latest commit to this branch uses the magic `auto` session. Because this feature is relatively new in systemd, the original lookup by PID is still used as a fallback.